cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -132,40 +128,27 @@ func StartTestServer(t *testing.T) (result *restclient.Config, tearDownForCaller
 // StartTestServerOrDie calls StartTestServer with up to 5 retries on bind error and dies with
 // t.Fatal if it does not succeed.
 func StartTestServerOrDie(t *testing.T) (*restclient.Config, TearDownFunc) {
-	// retry test because the bind might fail due to a race with another process
-	// binding to the port. We cannot listen to :0 (then the kernel would give us
-	// a port which is free for sure), so we need this workaround.
-
-	var err error
-
-	for retry := 0; retry < 5 && !t.Failed(); retry++ {
-		var config *restclient.Config
-		var td TearDownFunc
-
-		config, td, err = StartTestServer(t)
-		if err == nil {
-			return config, td
-		}
-		if err != nil && !strings.Contains(err.Error(), "bind") {
-			break
-		}
-		t.Logf("Bind error, retrying...")
+	config, td, err := StartTestServer(t)
+	if err == nil {
+		return config, td
 	}
 
 	t.Fatalf("Failed to launch server: %v", err)
 	return nil, nil
 }
 
-func freePort() int {
-	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+func createListenerOnFreePort() (net.Listener, int, error) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
what about IPv6? We had people running this on IPv6 only systems.

Reply:
I will replace  "127.0.0.1:0" with ":0"

Reply:
localhost does not work?